plugins/woocommerce/includes/tracks/class-wc-site-tracking.php

```diff
@@ -75,11 +75,38 @@ public static function add_tracking_function() {
 		 * @since 6.5.0
 		 */
 		$filtered_properties = apply_filters( 'woocommerce_tracks_event_properties', $client_tracking_properties, false );
+		$environment_type    = function_exists( 'wp_get_environment_type' ) ? wp_get_environment_type() : 'production';
 		?>
 		<!-- WooCommerce Tracks -->
 		<script type="text/javascript">
 			window.wcTracks = window.wcTracks || {};
 			window.wcTracks.isEnabled = <?php echo self::is_tracking_enabled() ? 'true' : 'false'; ?>;
+			window.wcTracks.validateEvent = function( eventName, props ) {
+				let isValid = true;
+				if ( ! <?php echo esc_js( WC_Tracks_Event::EVENT_NAME_REGEX ); ?>.test( eventName ) ) {
+					if ( <?php echo esc_attr( 'production' !== $environment_type ); ?> ) {
+						/* eslint-disable no-console */
+						console.error(
+							`A valid event name must be specified. The event name: "${ eventName }" is not valid.`
+						);
+						/* eslint-enable no-console */
+					}
+					isValid = false;
+				}
+				for ( const prop of Object.keys( props ) ) {
+					if ( ! <?php echo esc_js( WC_Tracks_Event::PROP_NAME_REGEX ); ?>.test( prop ) ) {
+						if ( <?php echo esc_attr( 'production' !== $environment_type ); ?> ) {
+							/* eslint-disable no-console */
+							console.error(
+								`A valid prop name must be specified. The property name: "${ prop }" is not valid.`
+							);
+							/* eslint-enable no-console */
+						}
+						isValid = false;
+					}
+				}
+				return isValid;
+			}
```

Comment:
I created this method to be used inside the method `recordEvent` (below) and in the component `Tracks` [here](https://github.com/woocommerce/woocommerce/pull/34005/files#diff-757275733e1ced98a36920c903285f880eaa01921d8aaf98b95652bb9076cdaaR46).
The downside about this is that this code won't be accessible by our [tests in the component](https://github.com/woocommerce/woocommerce/pull/34005/files#diff-fa904a8c5fdff070fca808572f7e0ee67050093be6ae352ac13445a6b32622e7R1-R37), but the good thing is that we have the code only in one place.